main.go

```diff
@@ -123,13 +122,14 @@ func generatorHandler(l zerolog.Logger) http.HandlerFunc {
 			return
 		}
 
-		// if withLatest is enabled we add a new release that is a copy of the
+		// if WithCustomTag is set, we add a new release that is a copy of the
 		// latest released version but with a `latest` name slug
-		if req.Input.Parameters.WithLatest {
+		if req.Input.Parameters.WithCustomTag != "" {
 			// filtered is an array of Release (not *Release) so this creates a copy
-			latest := filtered[len(filtered)-1]
-			latest.NameSlug = "latest"
-			latest.TagSlug = fmt.Sprintf("%s-latest", latest.TagSlug)
+			latest := Release{
```

Comment:
This change breaks the behavior of the plugin, we are returning something that is not a release of the project it is watching. The idea is to always return an output that is directly mapped to a list of releases that exists within the repositories itself. That is what I was doing before via `WithLatest`. It is a shorthand that is useful if some users don't want to always pin versions in a specific way. We could allow customizing the name, but I find that more confusing since it will always point to the latest release.

Reply:
Got it. In that case, what about leaving the plugin as is - closing this PR - and instead use the list generator for our purpose: https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-List/ . This is the end-goal that we are trying to achieve: https://github.com/dagger/dagger.io/pull/3798#issuecomment-2175930694

Reply:
Seeing our internal PR, I think I misunderstood what we were trying to achieve. I thought we were looking to have a `latest` tag that would point to the last release that was made. Seeing that we want main and (I assume) we plan to build the main image constantly, using the list generator + matrix is a much better approach